File: SecondSem/CN/LAB/Java/ConnectionLessFullDuplex_4/UDPServers.java
import java.net.*;

public class UDPServers {
    public static void main(String[] args) {
        try {
            DatagramSocket serverSocket = new DatagramSocket(9876);

            byte[] receiveData = new byte[1024];
            byte[] sendData = new byte[1024];

            System.out.println("Server started. Waiting for incoming messages...");

            while (true) {
                DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
                serverSocket.receive(receivePacket);

                String sentence = new String(receivePacket.getData(), 0, receivePacket.getLength());
                InetAddress IPAddress = receivePacket.getAddress();
                int port = receivePacket.getPort();

                System.out.println("Received from client at " + IPAddress + ":" + port + ": " + sentence);

                String capitalizedSentence = sentence.toUpperCase();
                sendData = capitalizedSentence.getBytes();

                DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, IPAddress, port);
                serverSocket.send(sendPacket);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: SecondSem/CN/LAB/Python/ConnectionLessFullDuplex_4/UDPservers.py
import socket


def main():
    try:
        server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        server_socket.bind(("", 9876))

        print("Server started. Waiting for incoming messages...")

        while True:
            receive_data, client_address = server_socket.recvfrom(1024)
            sentence = receive_data.decode("utf-8")
            ip_address, port = client_address

            print(f"Received from client at {ip_address}:{port}: {sentence}")

            capitalized_sentence = sentence.upper()
            send_data = capitalized_sentence.encode("utf-8")

            server_socket.sendto(send_data, client_address)
    except Exception as e:
        print(e)


if __name__ == "__main__":
    main()
